### ISSUE TITLE
Fail to run the example of coco_caption of caffe lstm 

### ISSUE THREAD

**Issue by liqing-ustc** [BVLC/caffe#2099]
I have successfully compiled the caffe with lstm from @jeffdonahue https://github.com/jeffdonahue/caffe/tree/recurrent, but I fail to run the example of coco_caption, the error is "F0311 22:45:02.564544 18964 blob.hpp:141] Check failed: num_axes() <= 4 (117452
vs. 4) Cannot use legacy accessors on Blobs with > 4 axes". @jeffdonahue Can you give me some tips?

**Comment by nian-liu** [comment]
Hi @liqing-ustc,
   I have the same problem with you, have you solved this issue? Any tip is appreciated.

**Comment by spandanagella** [comment]
Hi,
I'm having some issues with running lstm's with caffe. I succesfully compiled but not able to clear the runtests.

Error:
SoftmaxWithLossLayerTest/3.TestForwardIgnoreLabel
F0417 15:27:25.539069  9474 layer_factory.hpp:77] Check failed: registry.count(type) == 1 (0 vs. 1) Unknown layer type: Softmax 

Also, I tried to execute the LRCN model and LSTM. Both failed with the below error exception (LRCN and LSTM proto files from here https://github.com/jeffdonahue/caffe/blob/recurrent/examples/coco_caption/).

Error parsing text-format caffe.NetParameter: 52:15: Message type "caffe.LayerParameter" has no field named "embed_param".

Please let me know if someone has faced similar issues or any tips to solve this.

spandana

**Comment by niuchuang** [comment]
Could someone give me some guidance about how to construct a RNN with jeffdonahue's PR? I have downloaded the lrcn.prototxt , unfortunately I cannot understand most of its contents , such as include { stage: "freeze-convnet" }, include { stage: "unfactored" } and so on. In fact,I have some time sequence image data , each of which has a label. I have trained reference model in caffe with these data, and now I try to use RNN to classify them. What document I should read so that I can understand lrcn.prototxt and something like this,and then train a RNN model with my data. Much thanks !
